### PR TITLE
Feature: Add handleLLMNewFunctionCall callback

### PR DESCRIPTION
**Pull request by mgce**
Right now when we are using a function call, the `handleLLMNewToken` callback returns nothing. The `handleLLMNewFunctionCall` callback allows us to receive a streamed response.

### CODE REVIEW

File: langchain/src/callbacks/base.ts [hunk 1]
```diff
@@ -59,6 +59,17 @@ abstract class BaseCallbackHandlerMethodsClass {
     tags?: string[]
   ): Promise<void> | void;
 
+  /**
+   * Called when an LLM/ChatModel in `streaming` mode produces a new function call
+   */
+  handleLLMNewFunctionCall?(
+    function_call: { name?: string; arguments?: string },
```
Comment (jacoblee93) [inline langchain/src/callbacks/base.ts:66]: nit: Prefer camel case

Comment (mgce) [reply langchain/src/callbacks/base.ts]: Changed

**Comment by jacoblee93** [comment]
Thanks for the PR - had some discussion with some of the Python maintainers, and it sounds like the preference would be for something more generic like a catchall `handleEvent`. Will leave this open for discussion for now.

**Comment by mgce** [comment]
@jacoblee93 Thanks for the reply. I have created a follow-up PR including this suggestion. 
https://github.com/hwchase17/langchainjs/pull/2025

**Comment by jacoblee93** [comment]
Wow thank you! Will close this one then.

**Pull request closed without merging**